src/app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-cluster.cpp

```diff
@@ -0,0 +1,372 @@
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "wifi-network-diagnostics-cluster.h"
+
+#include <app/EventLogging.h>
+#include <clusters/WiFiNetworkDiagnostics/Attributes.h>
+#include <clusters/WiFiNetworkDiagnostics/Commands.h>
+#include <clusters/WiFiNetworkDiagnostics/Ids.h>
+#include <clusters/WiFiNetworkDiagnostics/Metadata.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <tracing/macros.h>
+#include <tracing/metric_event.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::WiFiNetworkDiagnostics;
+using namespace chip::app::Clusters::WiFiNetworkDiagnostics::Attributes;
+using chip::DeviceLayer::DiagnosticDataProvider;
+using chip::DeviceLayer::GetDiagnosticDataProvider;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+template <typename T, typename Type>
+CHIP_ERROR WiFiDiagnosticsServerLogic::ReadIfSupported(CHIP_ERROR (DeviceLayer::DiagnosticDataProvider::*getter)(T &), Type & data,
+                                                       AttributeValueEncoder & aEncoder)
+{
+    T value;
+    CHIP_ERROR err = (mDiagnosticProvider.*getter)(value);
+
+    if (err == CHIP_NO_ERROR)
+    {
+        data.SetNonNull(value);
+    }
+    else
+    {
+        ChipLogProgress(Zcl, "The WiFi interface is not currently configured or operational.");
+    }
+
+    return aEncoder.Encode(data);
+}
+
+CHIP_ERROR WiFiDiagnosticsServerLogic::ReadWiFiBssId(AttributeValueEncoder & aEncoder)
+{
+    Attributes::Bssid::TypeInfo::Type bssid;
+
+    uint8_t bssidBytes[chip::DeviceLayer::kMaxHardwareAddrSize];
+    MutableByteSpan bssidSpan(bssidBytes);
+    if (mDiagnosticProvider.GetWiFiBssId(bssidSpan) == CHIP_NO_ERROR)
+    {
+        if (!bssidSpan.empty())
+        {
+            bssid.SetNonNull(bssidSpan);
+            ChipLogProgress(Zcl, "Node is currently connected to Wi-Fi network with BSSID:");
+            ChipLogByteSpan(Zcl, bssidSpan);
+        }
+    }
+    else
+    {
+        ChipLogProgress(Zcl, "The WiFi interface is not currently connected.");
+    }
+
+    return aEncoder.Encode(bssid);
+}
+
+CHIP_ERROR WiFiDiagnosticsServerLogic::ReadSecurityType(AttributeValueEncoder & aEncoder)
+{
+    Attributes::SecurityType::TypeInfo::Type securityType;
+    SecurityTypeEnum value = SecurityTypeEnum::kUnspecified;
+
+    if (mDiagnosticProvider.GetWiFiSecurityType(value) == CHIP_NO_ERROR)
```

Comment:
Is there a reason this is not using ReadIfSupported?  If so, it should probably be documented.  Similar for some of these other attributes.

Reply:
Added documentation in the header file.